README.md: add Home Assistant and ESPHome requirements to BLE proxy description (#715)
matter-js/matterjs-server: README.md

bridges BLE traffic from wherever the adapter actually lives. See
[`docs/ble-proxy-protocol.md`](docs/ble-proxy-protocol.md) for the protocol spec.

> For BLE proxy mode to work, you need:

Reviewer: ```suggestion
When using the BLE proxy mode together with Home Assistant, you need:
```

Author: Is there a way to use BLE proxy *without* Home Assistant? It's my understanding that this is not possible, as BLE proxies are a feature of HA.

Reviewer: The Matter server implements a generic BLE Websocket interface and a python client to connect "python bluetooth logic to that". HA then has logic to integrate the ESPHome BLE Proxy feature with that python bluetooth stuff. SO the server is generic here

Author: Ok, I applied the change.

> When using the BLE proxy mode together with Home Assistant, you need:

Reviewer: ```suggestion
When using the BLE proxy mode together with Home Assistant (which uses the ble-prox-python -client), you need:
```

> When using the BLE proxy mode together with Home Assistant (which uses the ble-prox-python -client), you need:

Reviewer: ```suggestion
When using the BLE proxy mode together with Home Assistant (which uses the ble-proxy-python -client), you need:
```

- Home Assistant 2026.6 (or later)
- ESPHome 2026.5 (or later)

> - esp-idf 6.0.1 (exactly)

Reviewer: why exactly and not "or higher"?

Author: Because at the moment the necessary fix is only on 6.0.1. We can adapt the wording once the fix gets ported to other branches. (I hope it will be ported to 5.5.x, the current default.)

Reviewer: ok but it will also stay in 6.02 I guess :-) so I think 6.0.1+ is the better wording?

Author: Well, there is no 6.0.2 yet. Only a 6.1-dev, and the fix isn't in there. On top of that, ESPHome at the moment only supports 6.0.1 not 6.0.2 or 6.1-dev. So I think until the situation is cleared up, 6.0.1 is the better wording. We can change it later.

> - esp-idf 6.0.1 (exactly)

Reviewer: ```suggestion
- esp-idf 6.0.1 (or later)
```
